cmake/FetchMongoC.cmake

```
@@ -2,43 +2,56 @@
 
 include(FetchContent)
 
-message(STATUS "Download and configure C driver version ${LIBMONGOC_DOWNLOAD_VERSION} ... begin")
-
-set(fetch_args "")
-if (CMAKE_VERSION VERSION_GREATER_EQUAL "3.25.0")
-    set(fetch_args "SYSTEM")
-endif()
-
-# Declare mongo-c-driver as a dependency
-FetchContent_Declare(
-    mongo-c-driver
-    GIT_REPOSITORY https://github.com/mongodb/mongo-c-driver.git
-    GIT_TAG ${LIBMONGOC_DOWNLOAD_VERSION}
-
-    ${fetch_args}
-)
-
-FetchContent_GetProperties(mongo-c-driver)
-
-if(NOT mongo-c-driver_POPULATED)
-    set(OLD_ENABLE_TESTS ${ENABLE_TESTS})
-    set(OLD_BUILD_TESTING ${BUILD_TESTING})
-    set(OLD_CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS})
-    set(OLD_CMAKE_C_FLAGS ${CMAKE_C_FLAGS})
-
-    # Set ENABLE_TESTS to OFF to disable the test-libmongoc target in the C driver.
-    # This prevents the LoadTests.cmake script from attempting to execute test-libmongoc.
-    # test-libmongoc is not built with the "all" target.
-    # Attempting to execute test-libmongoc results in an error: "Unable to find executable: NOT_FOUND"
-    set(ENABLE_TESTS OFF)
-    set(BUILD_TESTING OFF)
-    string(REPLACE " -Werror" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
-    string(REPLACE " -Werror" "" CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
-    FetchContent_MakeAvailable(mongo-c-driver)
-    set(CMAKE_CXX_FLAGS ${OLD_CMAKE_CXX_FLAGS})
-    set(CMAKE_C_FLAGS ${OLD_CMAKE_C_FLAGS})
-    set(ENABLE_TESTS ${OLD_ENABLE_TESTS})
-    set(BUILD_TESTING ${OLD_BUILD_TESTING})
-endif()
-
-message(STATUS "Download and configure C driver version ${LIBMONGOC_DOWNLOAD_VERSION} ... end")
+function(fetch_mongoc)
+    message(STATUS "Download and configure C driver version ${LIBMONGOC_DOWNLOAD_VERSION} ... end")
+
+    set(fetch_args "")
+    if (CMAKE_VERSION VERSION_GREATER_EQUAL "3.25.0")
+        list(APPEND fetch_args "SYSTEM")
+    endif()
+
+    # Declare mongo-c-driver as a dependency
+    FetchContent_Declare(
+        mongo-c-driver
+        GIT_REPOSITORY https://github.com/mongodb/mongo-c-driver.git
+        GIT_TAG ${MONGOC_DOWNLOAD_VERSION}
```

```suggestion
        GIT_TAG ${LIBMONGOC_DOWNLOAD_VERSION}
```

This may fix failing tasks that seem to be using the auto-downloaded C driver. [Example](https://spruce.mongodb.com/task/mongo_cxx_driver_versioned_api_matrix_versioned_api_two_accepted_macos_14_arm64_debug_shared_patch_50cda18f9233e15bb0daf90715bc586db7650499_67f406375617eb000769ff23_25_04_07_17_07_08/logs?execution=0).